cdap-ui/app/cdap/components/DataPrep/store/index.ts

```diff
@@ -18,12 +18,65 @@ import { combineReducers, createStore } from 'redux';
 import DataPrepActions from 'components/DataPrep/store/DataPrepActions';
 import { composeEnhancers } from 'services/helpers';
 
-const defaultAction = {
+export interface IDataPrepAction {
+  type: string;
+  action?: string;
+  payload?: any;
+}
+
+const defaultAction: IDataPrepAction = {
+  type: '',
   action: '',
   payload: {},
 };
 
-const defaultInitialState = {
+export interface IDataModel {
+  uuid: string;
+  id: string;
+  revision: number;
+  name: string;
+  description?: string;
+  url?: string;
+  models?: IModel[];
+}
+
+export interface IModel {
+  uuid: string;
+  id: string;
+  name: string;
+  description?: string;
+  fields?: IModelField[];
+}
+
+export interface IModelField {
+  uuid: string;
+  id: string;
+  name: string;
+  description?: string;
+}
+
+export interface IDataPrepState {
+  initialized?: boolean;
+  workspaceId?: string;
+  workspaceUri?: string;
+  data?: any;
+  headers?: any;
+  types?: any; // pure column types from backend. Used for display
+  typesCheck?: any; // case sensitive column types, Should be used when checking types of column
+  selectedHeaders?: any;
```

Comment:
Seems like this interface is having too many `any` types. Do we know the structure of the response? Using this many `any`s just takes away the point of using typescript though.

Reply:
I can only guess on types because I don't know exact structure. I can extract types by looking at response data.

Previously you asked to add types to only modified fields:

> Should be good to add types for only those things that are being modified, for instance dataprep store

Reply:
Ok. If those are the fields that are not used by map-to-target directive then its ok leave it as is. But please a TODO to add types for unknown ones.

Reply:
Done.